docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-utils-advanced-operation/index.md

```diff
@@ -7,3 +7,336 @@ sidebar_position: 500
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__user_identifiers: [{'schema': 'com_mycompany_user_identifier_2_0_0', 'field': 'new_user_id', 'prefix': 'ui_t'}, {'schema': 'com_mycompany_user_identifier_1_0_0', 'field': 'user_id', 'prefix': 'ui_o'}]
+    ...
+...
+```
+This setup implies that the `new_user_id` from `com_mycompany_user_identifier_2_0_0` should have precedence over the `user_id` field from `com_mycompany_user_identifier_1_0_0`, meaning that if both are filled for a particular event, the `new_user_id` value would be the one present in the `user_identifier` field in your dbt tables. If you'd prefer to have the precedence swapped, you can swap the ordering in the `dbt_project.yml`.
+
+Handling minor version bumps of your schemas where this isn't automatically handled by your loader works in the exact same way. A small example of bumping `com_mycompany_user_identifier_1_0_0` to `com_mycompany_user_identifier_1_1_0` but leaving the `user_id` field as the key identifier would look as follows:
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__user_identifiers: [{'schema': 'com_mycompany_user_identifier_1_1_0', 'field': 'user_id', 'prefix': 'ui_t'}, {'schema': 'com_mycompany_user_identifier_1_0_0', 'field': 'user_id', 'prefix': 'ui_o'}]
+    ...
+...
+```
+:::info
+Remember that the `prefix` key only needs to be set when running these models on Postgres or Redshift.
+:::
+
+### Combining atomic fields and custom contexts
+Combining atomic fields and custom contexts should hopefully then be straightforward if you're comfortable with what we described above. Let's say you want to combine the `logged_in_user_id` field from the `com_mycompany_logged_user_id_1_0_0` context together with the standard `domain_userid` field in the `events` table. To achieve that, you would include the following in your `dbt_profile.yml`:
+
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__user_identifiers: [{'schema': 'com_mycompany_logged_user_id_1_0_0', 'field': 'logged_in_user_id', 'prefix': 'lui'}, {'schema': 'atomic', 'field': 'domain_userid', 'prefix': 'e'}]
+    ...
+...
+```
+
+This would be parsed into the following SQL:
+
+```sql
+SELECT
+    ...
+    COALESCE(lui_logged_in_user_id, domain_userid, NULL) as user_identifier,
+...
+```
+# Introducing custom SQL logic to every event
+If there are certain SQL transformations you want to apply to events that are being processed by Snowplow's dbt packages you can leverage the `snowplow__custom_sql` variable to write out custom SQL that will be included in your `base_events_this_run` table, which can then be leveraged for any of your subsequent tables.
```

Comment:
It might be worth adding in the section an example of the macro where you use it as a "with base as ( {{macro}}),..." to add more complex post-processing, as we do in the web package with it now, as that might be easier for people than using these variables in some cases?